lib/buildkite/test_collector/minitest_plugin/trace.rb

```diff
@@ -72,17 +70,20 @@ def project_dir
     end
 
     def failure_reason
-      @failure_reason ||= example.failure&.message
+      @failure_reason ||= strip_invalid_utf8_chars(example.failure&.message)&.split("\n")&.first
     end
 
     def failure_expanded
-      @failure_expanded ||= begin
-        example.failures.map do |failure|
-          {
-            expanded: failure.message,
-            backtrace: failure.backtrace,
-          }
-        end
+      @failure_expanded ||= example.failures.map.with_index do |failure, index|
+        # remove the first line of message from the first failure
+        # to avoid duplicate line in Test Analytics UI
+        messages = strip_invalid_utf8_chars(failure.message).split("\n")
+        messages = messages[1..] if index.zero?
+
+        {
+          expanded: messages,
```

Comment:
It looks like we are sending an array of strings rather than a string. Is that the intended behaviour?

Reply:
That is what the rspec collector and other collectors are sending (i.e. array of strings). The UI also expects it to be an array of strings, and sending a string will break it. The documentation doesn't explicitly say anything about the expected type for `expanded` but the example has an array of strings. https://buildkite.com/docs/test-analytics/importing-json#json-test-results-data-reference

Reply:
So it was likely an error/bug before this change?

Reply:
I won't say it is a bug in the test collector itself because the documentation for `failure_expanded` doesn't specify the expected type. Maybe it is the UI that needs to be more flexible. 

@meghan-kradolfer since you've been working on this area recently, do you have an opinion on this?


Key | Type | Description
-- | -- | --
failure_expanded | array of hashes | A more detailed explanation of why the test failed

Reply:
I wouldn't say that this is an error/bug, but perhaps a quirk of the rspec test collector being designed first, resulting in our data being quite dictated by the shape and design of rspec. 

For a failing test, rspec generates an output that looks like this:
```
     TypeError:
       no implicit conversion of Array into String
     # ./app/workers/analytics/fetch_artifact_for_ingestion_worker.rb:22:in `initialize'
     # ./app/workers/analytics/fetch_artifact_for_ingestion_worker.rb:22:in `new'
     # ./app/workers/analytics/fetch_artifact_for_ingestion_worker.rb:22:in `perform'
     # ./spec/workers/analytics/fetch_artifact_for_ingestion_worker_spec.rb:89:in `block (3 levels) in <top (required)>'
     # ./spec/support/log.rb:17:in `run'
     # ./spec/support/log.rb:66:in `block (2 levels) in <top (required)>'
     # ./spec/support/database.rb:19:in `block (2 levels) in <top (required)>'
```

The first two lines are the "expanded" and the lines starting with `#` become "backtrace". You can see that the [message lines](https://github.com/buildkite/test-collector-ruby/blob/6420d36f5cdb3e5f7888690803e54c3b452fde4b/lib/buildkite/test_collector/rspec_plugin/reporter.rb#L63) that become the data in "expanded" are an array of strings, so we've just replicated that structure. 

I'm not saying that just doing what rspec does is a good pattern to adopt across all our test collectors in different languages, but seeing as the data shown here is basically log lines, I do think having an array of strings is possibly the most straight forward in terms of formatting on the front end? Maybe its annoying to enforce this array of strings format on the collector side, but it seems cleaner than having to parse and split outputs from a variety of different frameworks.

Reply:
I created a post regarding `failure_expanded` on our Basecamp, it will be good to discuss this issue separately.